manager/misc.go

```diff
@@ -75,30 +75,44 @@ func (m *VolumeManager) GenerateSupportBundle() (io.ReadCloser, string, int64, e
 	bundleName := "longhorn-support-bundle_" + bundleMeta.LonghornNamespaceUUID + "_" +
 		strings.Replace(bundleMeta.BundleCreatedAt, ":", "-", -1)
 	bundleFileName := bundleName + ".zip"
-	bundleDir := filepath.Join("/tmp", bundleName)
-	bundleFile := filepath.Join("/tmp", bundleFileName)
 
-	if err := os.MkdirAll(bundleDir, os.FileMode(0755)); err != nil {
-		return nil, "", 0, err
-	}
-	m.generateSupportBundle(bundleDir, bundleMeta)
+	go func() {
+		bundleDir := filepath.Join("/tmp", bundleName)
+		bundleFile := filepath.Join("/tmp", bundleFileName)
+		if err := os.MkdirAll(bundleDir, os.FileMode(0755)); err != nil {
+			m.sb.Error = BundleMkdirFailed
+			m.sb.State = BundleStateError
+			return
+		}
+		m.generateSupportBundle(bundleDir, bundleMeta)
+		cmd := exec.Command("zip", "-r", bundleFileName, bundleName)
+		cmd.Dir = "/tmp"
+		if err := cmd.Run(); err != nil {
+			m.sb.Error = BundleZipFailed
+			m.sb.State = BundleStateError
+			return
+		}
+		f, err := os.Open(bundleFile)
```

Comment:
Change to use `os.Stat`.

And the file isn't closed if the go routine errors out.